docs/pages/database-access/guides/mongodb-self-hosted.mdx

```diff
@@ -0,0 +1,400 @@
+---
+title: Database Access with Self-Hosted MongoDB
+description: How to configure Teleport Database Access with self-hosted MongoDB
+---
+
+# Self-Hosted MongoDB
+
+<Admonition type="note" title="Supported versions">
+  Teleport Database Access supports MongoDB 3.6 and higher. Older versions have
+  not been tested and are not guaranteed to work. MongoDB 3.6 was released in
+  November 2017 and reached EOL in April 2021 so if you're still using an older
+  version, consider upgrading.
+</Admonition>
+
+## Create Certificate/Key Pair
+
+Teleport uses mutual TLS authentication with MongoDB. As such, self-hosted
+MongoDB cluster must be configured with Teleport's certificate authority and a
+certificate/key pair that Teleport can validate.
+
+To create these secrets, use `tctl auth sign` command. It can be run either
+[locally](../../architecture/overview.mdx#tctl) on the auth server, or from the
+client machine after logging in with `tsh login` (for example, when using
+[Teleport Cloud](../../cloud/introduction.mdx)).
+
+In the latter case, user must be allowed to impersonate the system role `Db` in
+order to be able to generate the database certificate, by having the following
+allow rule in their role:
+
+```yaml
+allow:
+  impersonate:
+    users: ["Db"]
+    roles: ["Db"]
+```
+
+Create the secrets:
+
+<Tabs>
+  <TabItem label="Standalone">
+  When connecting to standalone MongoDB, sign certificate for the hostname over
+  which Teleport will be connecting to it.
+
+  For example, if your MongoDB server is accessible at `mongo.example.com`
+  hostname, run:
+
+  ```bash
+  $ tctl auth sign --format=mongodb --host=mongo.example.com --out=mongo --ttl=8760h
+  ```
+
+  The command will create 2 files: `mongo.cas` with Teleport's certificate
+  authority and `mongo.crt` with generated certificate and key pair. You will
+  need these files to enable mutual TLS on your MongoDB server.
+  </TabItem>
+  <TabItem label="Replica set">
+  When connecting to a MongoDB replica set, sign certificates for each member
+  using the hostnames they're accessible at.
+
+  For example, if the first member is accessible at `mongo1.example.com` and
+  the second at `mongo2.example.com`, run:
+
+  ```bash
+  $ tctl auth sign --format=mongodb --host=mongo1.example.com --out=mongo1 --ttl=8760h
+  $ tctl auth sign --format=mongodb --host=mongo2.example.com --out=mongo2 --ttl=8760h
+  ```
+
+  Each command will create 2 files: `mongo{1,2}.cas` with Teleport's certificate
```

Comment:
```suggestion
  Each command will create 2 files: `mongo1.cas` and `mongo2.cas` with Teleport's certificate
```

Might be better to be explicit about the filenames here - some people may not be familiar with the `bash` `{1,2}` syntax format.